sopel/bot.py

```diff
@@ -324,13 +329,20 @@ def remove_plugin(self, plugin, callables, jobs, shutdowns, urls):
             self.unregister(func)
 
         # remove URL callback handlers
+        for func in urls:
+            regexes = func.url_regex
+            for regex in regexes:
+                if func in self._url_callbacks.get(regex, []):  # can .get ever fail?
+                    self.unregister_url_callback(regex, func)
+                    LOGGER.debug('URL Callback unregistered: %r', regex)
+        # and the old way, until 8.0
         if "url_callbacks" in self.memory:
             for func in urls:
                 regexes = func.url_regex
                 for regex in regexes:
                     if func == self.memory['url_callbacks'].get(regex):
                         self.unregister_url_callback(regex, func)
-                        LOGGER.debug('URL Callback unregistered: %r', regex)
+                        LOGGER.warning('Legacy URL Callback unregistered: %r', regex)
```

Comment:
I believe this is the right place and the right amount of warning. Nothing more needed.  :+1: